### PR TITLE
GEODE-5185: Correct backwards-compatibility violation in 'create gateway-receiver'

### PR DESCRIPTION
**Pull request by PurelyApplied**
* Use new ResultModel and ClusterConfiguration designs.
* Add explicit message when persistance fails due to command error.

-----

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by PurelyApplied** [comment]
Two things:

* Though beyond the scope of this ticket, I think the addition of an explicit "This hasn't persisted" in the event of error would be a good addition to the recent changes surrounding the `SingleGfshCommand`.
* To limit scope, I don't actually pass the `GatewayReceiver` configuration object to the members, but dissect it for consumption by the existing framework.  I could go either way on whether or not the configuration object itself should be the core of the function arguments.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java [hunk 1]
```diff
@@ -80,46 +83,71 @@ public Result createGatewayReceiver(@CliOption(key = {CliStrings.GROUP, CliStrin
...
+    }
+    if (socketBufferSize != null) {
+      configuration.setSocketBufferSize(String.valueOf(socketBufferSize));
+    }
+    configuration.setHostnameForSenders(hostnameForSenders);
+    configuration.setManualStart(manualStart);
+    return configuration;
+  }
+
   public static class Interceptor extends AbstractCliAroundInterceptor {
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java]: When switching to `ResultModel` you'll need to change any `Interceptor`s too. `preExecution` should return a `ResultModel` and `postExecution` a `CommandResult`. Now would be a good time to add a test for these :). Typically `preExecution` is just validation - a simple approach for testing can be seen in `ExecuteFunctionCommandTest`.

Comment (jdeppe-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java]: That example is still pending review...

Comment (PurelyApplied) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java]: Oooo, good point.

Looks like the `preExecution` is already checked in the UnitTest, but yeah, good to double check for potential gaps in our coverage.

**Comment by PurelyApplied** [comment]
I've added `GEODE-5185` to the head of this PR, since they both touch the same command.

The commit message should probably prioritize that ticket rather than the refactor, to something like
```
GEODE-5185: Correct backwards-compatibility violation in 'create gateway-receiver' command.
* Also refactored command per GEODE-4858 and GEODE-5010 to use the SingleGfshCommand and ResultModel classes, respectively.
```

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java [hunk 2]
```diff
@@ -80,49 +81,75 @@ public Result createGatewayReceiver(@CliOption(key = {CliStrings.GROUP, CliStrin
...
-    // no xml needs to be updated, simply return
-    if (xmlEntity == null) {
-      return result;
-    }
+    ResultModel result = ResultModel.createMemberStatusResult(gatewayReceiverCreateResults);
 
-    boolean allSuccessful = gatewayReceiverCreateResults.stream()
-        .map(CliFunctionResult::isSuccessful).reduce(true, (x, y) -> x && y);
-    if (!allSuccessful) {
+    boolean anySuccessful = gatewayReceiverCreateResults.stream()
```
Comment (jinmeiliao) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java]: Is this block still necessary? I think ResultModel.ceateMemberStatusResult already sets the flag.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java [hunk 3]
```diff
@@ -109,6 +109,9 @@ protected Object invokeCommand(Object command, GfshParseResult parseResult) {
     SingleGfshCommand gfshCommand = (SingleGfshCommand) command;
     ResultModel resultModel = (ResultModel) result;
     if (resultModel.getStatus() == Result.Status.ERROR) {
+      InfoResultModel errorInfo = resultModel.addInfo();
```
Comment (jinmeiliao) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: I think we probably don't need this. If the command already failed. No one would be expecting the cluster config be changed.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java (hunk 2 above)
Comment (PurelyApplied) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java]: Oh, duh.  Good catch.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java (hunk 3 above)
Comment (PurelyApplied) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: Wasn't so sure of it myself.  I'll pull it out.

**Comment by PurelyApplied** [comment]
Precheckin instability kept its head down for long enough for one to come back green!  Huzzah.

**Pull request merged**